### PR TITLE
proxy: vm: Improve hyperstart protocol abstraction

### PR DESCRIPTION
**Pull request by sboeuf**
This PR fixes issue #439.
It allows the proxy to rely on latest API defined by the hyperstart package, so that it does not need to be aware about any details of the hyperstart protocol.

### CODE REVIEW

**Comment by sboeuf** [comment]
We need to wait for PR 65 in https://github.com/sameo/virtcontainers/pull/65 to be merged, before we can proceed with this one. Travis won't build properly until it is merged in virtcontainers.

**Comment by sameo** [comment]
The needed hyperstart package changes have been merged into virtcontainers [v0.2.0](https://github.com/sameo/virtcontainers/releases/tag/v0.2.0)

**Comment by sameo** [comment]
@sboeuf You may want to add the VirtContainers vendoring (to v0.2.0) as part of this PR.

**Comment by sboeuf** [comment]
@sameo I agree I should vendor this to version 0.2.0 but I will wait for PR #468 to be first merged. After that, I will only need to change the packages.json file.

**Comment by sboeuf** [comment]
PR #468 has been merged, meaning the vendoring for virtcontainers  has been introduced. We can now merge this one (relies on virtcontainers v0.2.0).

**Comment by dlespiau** [comment]
It's not enough to change the version in packages.json :) vendoring works by copying the corresponding files in the vendor/ directory ; that's why tools like ciao-vendor or govendor (https://github.com/kardianos/govendor) exist.

**Comment by sboeuf** [comment]
That's now vendored properly. It adds the README.md modfications and the hyperstart package modification bringing new API.

File: proxy/vm.go [hunk 1]
```diff
@@ -173,19 +167,19 @@ func (vm *vm) SendMessage(cmd string, data []byte) error {
 // There's one instance of this goroutine per client having done an allocateIO.
 func (vm *vm) ioClientToHyper(session *ioSession) {
 	for {
-		seq, data, err := readIoMessage(session.client)
```
Comment (dlespiau) [inline proxy/vm.go]: the whole readIoMessage() function can be removed then?

**Comment by chavafg** [comment]
qa-passed

**Comment by dlespiau** [comment]
lgtm

**Comment by dlespiau** [comment]
(there's a conflict to solve though, needs rebasing)

**Comment by sboeuf** [comment]
Just rebased. Let's see if travis still passing !

**Comment by dlespiau** [comment]
nop :)

**Comment by sboeuf** [comment]
New try !

**Comment by jodh-intel** [comment]
lgtm.

**Comment by dlespiau** [comment]
QA and lgtms, a wall of green!

**Pull request merged**